autopeering/discover/protocol.go

```diff
@@ -32,36 +34,33 @@ var retryPolicy = backoff.ExponentialBackOff(500*time.Millisecond, 1.5).With(
 type Protocol struct {
 	server.Protocol
 
-	loc *peer.Local    // local peer that runs the protocol
-	log *logger.Logger // logging
+	version uint32         // specifies the expected version number for this Protocol
```

Comment:
Should we explicitly split the `version` field into networkID and version? WDYT?

Reply:
Yes :)

Reply:
So have a `uint32 version` and a `string networkID` or what are you proposing?

Reply:
the version field could still be there just for defining the protocol version. Maybe 1 byte is enough for that? 256 possible different versions? For the networkID I like the idea of using string but it should be short and of fixed size